runtime/lib/backend/null_qubit/NullQubit.hpp

```diff
@@ -94,6 +137,27 @@ struct NullQubit final : public Catalyst::Runtime::QuantumDevice {
     {
         num_qubits_ = 0;
         this->qubit_manager.ReleaseAll();
+        if (this->track_resources_) {
+            auto time = std::chrono::high_resolution_clock::now();
+            auto timestamp =
+                std::chrono::duration_cast<std::chrono::nanoseconds>(time.time_since_epoch())
+                    .count();
+            std::stringstream resources_fname;
+            resources_fname << "__pennylane_resources_data_" << timestamp << ".json";
+
+            // Need to use FILE* instead of ofstream since ofstream has no way to atomically open a
+            // file only if it does not already exist
+            FILE *resources_file = fopen(resources_fname.str().c_str(), "wx");
+            if (resources_file == nullptr) {
+                std::cerr << "Error opening file: " << resources_fname.str();
+                // TODO: Should this throw?
```

Comment:
Great question, in a scenario where there are potentially many executions that print out resources we might as well exit early.

Alternately, if we still want to provide that info to the user you could fall-back to stdout.

I think given that this is to be used in CI, we should raise an exception so that wrong data isn't collected and reported silently.

Reply:
Fixed in [3794c2f](https://github.com/PennyLaneAI/catalyst/pull/1619/commits/3794c2f900ff47f425deb457f35c014b74650a7d)